cylc/flow/prerequisite.py

```diff
@@ -211,7 +221,7 @@ def set_conditional_expr(self, expr):
                 # Use '\b' in case one task name is a substring of another
                 # and escape special chars ('.', timezone '+') in task IDs.
                 expr = re.sub(
-                    fr"\b{re.escape(self.MESSAGE_TEMPLATE % t_output)}\b",
+                    fr"{re.escape(self.MESSAGE_TEMPLATE % t_output)}\b",
```

Comment:
TODO: consider the preceding comment - might need a more sophisticated fix?

Reply:
(Tests all pass, but maybe the commented issue only tested the other word boundary for task name sub-strings)

Reply:
I've pushed this fix and it reveals that your fix breaks this case